src/test/rgw/amqp_mock.cc

```diff
@@ -120,6 +126,35 @@ amqp_socket_t* amqp_tcp_socket_new(amqp_connection_state_t state) {
   return state->socket;
 }
 
+amqp_socket_t* amqp_ssl_socket_new(amqp_connection_state_t state) {
+  state->socket = new amqp_socket_t;
+  state->use_ssl = true;
+  return state->socket;
+}
+
+int amqp_ssl_socket_set_cacert(amqp_socket_t *self, const char *cacert) {
+  // do nothing
+  return AMQP_STATUS_OK;
+}
+
+void amqp_ssl_socket_set_verify_peer(amqp_socket_t *self, amqp_boolean_t verify) {
+  // do nothing
+}
+
+void amqp_ssl_socket_set_verify_hostname(amqp_socket_t *self, amqp_boolean_t verify) {
+  // do nothing
+}
+
+#if AMQP_VERSION >= AMQP_VERSION_CODE(0, 10, 0, 1)
+SSL_CTX* amqp_ssl_socket_get_context(amqp_socket_t *self) {
```

Comment:
```suggestion
void* amqp_ssl_socket_get_context(amqp_socket_t *self) {
```
gcc10 complains about ambiguity

Reply:
Fixed